src/charm.py

```diff
@@ -113,7 +113,12 @@ def __init__(self, *args):
             self._ports.api,
             external_url=lambda: AlertmanagerCharm._external_url.fget(self),  # type: ignore
         )
-        self.api = Alertmanager(port=self._ports.api, web_route_prefix=self.web_route_prefix)
+
+        self.api = Alertmanager(
+            port=self._ports.api,
+            web_route_prefix=self.web_route_prefix,
+            scheme=urlparse(self._external_url).scheme,
+        )
```

Comment:
I wonder if we should switch from parts to full url:
```suggestion
        self.api = Alertmanager(self._external_url)
```